### ISSUE TITLE
Multiple assignment has different effects for struct members and local variables

### ISSUE THREAD

**Issue by KytoDragon** [ldc-developers/ldc#2905]
```D
struct Foo {
    int x;

    this(int dummy) {
        x = dummy;
    }
}

struct Bar {
    Foo a;
    Foo b;

    this(int dummy) {
        a = (b = Foo(dummy));
    }
}

void main() {
    Foo a;
    Foo b;
    a = (b = Foo(7));
    assert(b.x == 7 && a.x == 7); // works
    Bar bar = Bar(7);
    assert(bar.b.x == 7 && bar.a.x == 7); // fails, bar.b.x is 0 !
}
```

Is there a concrete reason why these two results differ?
Using LDC2 1.12.0 an Windows 7 x64.

**Comment by kinke** [comment]
That's an interesting one. It's not LDC-specific, it's a front-end issue, so please file an issue with DMD.

**Comment by kinke** [comment]
According to run.dlang.io, this has worked from v2.060 to 2.069, and fails since 2.070.

**Comment by KytoDragon** [comment]
Created DMD issue 19389: https://issues.dlang.org/show_bug.cgi?id=19389 .

**Comment by kinke** [comment]
Was fixed upstream in 2.084 and so working with LDC 1.14.
